packages/nameguard-js/src/graphemes.ts

```diff
@@ -1,44 +1,578 @@
+  "\u113b",
+  "\u1123",
+  "\u317c",
+  "\u114d",
+  "\u11f1",
+  "\u3137",
+  "\u1117",
+  "\ud7c9",
+  "\u3136",
+  "\u318c",
+  "\u115b",
+  "\ud7db",
+  "\u1132",
+  "\u1134",
+  "\u11de",
+  "\u11cd",
+  "\ua96d",
+  "\u119d",
+  "\u313c",
+  "\u318e",
+  "\u318b",
+  "\u11a6",
+  "\u113d",
+  "\ua967",
+  "\u3130",
+  "\u11e7",
+  "\u11a2",
+  "\ua961",
+  "\ud7b8",
+  "\u11c4",
+  "\u11b9",
+  "\u315b",
+  "\u1149",
+  "\u3179",
+  "\ud7f7",
+  "\u11c1",
+  "\ud7be",
+  "\u119b",
+  "\u116e",
+  "\u3174",
+  "\u1142",
+  "\ua979",
+  "\ud7c1",
+  "\u1186",
+  "\u318d",
+  "\ua966",
+  "\u1196",
+  "\u113e",
+  "\u3150",
+  "\u1151",
+  "\ud7fc",
+  "\u318f",
+  "\u314b",
+  "\u11d6",
+  "\u116a",
+  "\ud7c5",
+  "\u3143",
+  "\u11c6",
+  "\u1124",
+  "\u3171",
+  "\u117c",
+  "\u119f",
+  "\ua96c",
+  "\ud7d1",
+  "\u3131",
+  "\u3132",
+  "\u11f2",
+  "\u11da",
+  "\u316f",
+  "\u1112",
+  "\u3181",
+  "\ud7f2",
+  "\ud7f9",
+  "\u11b0",
+  "\u1111",
+  "\u3186",
+  "\ud7ff",
+  "\u114b",
+  "\u112a",
+  "\u11e4",
+  "\ud7d4",
+  "\ud7b9",
+  "\u1101",
+  "\ud7cf",
+  "\ud7dd",
+  "\u11d9",
+  "\u11d7",
+  "\u1159",
+  "\u11ba",
+  "\ud7c0",
+  "\u118c",
+  "\u11fa",
+  "\u11d1",
+  "\u1173",
+  "\u1129",
+  "\u1100",
+  "\ud7f0",
+  "\u1110",
+  "\ud7ce",
+  "\u1130",
+  "\u3165",
+  "\u3188",
+  "\u3160",
+  "\u3158",
+  "\u1103",
+  "\u111a",
+  "\ud7c8",
+  "\ud7b2",
+  "\u1198",
+  "\u11ce",
+  "\u115c",
+  "\ud7da",
+  "\u1121",
+  "\ud7e8",
+  "\ud7d7",
+  "\u3139",
+  "\ua963",
+  "\u114c",
+  "\ud7ee",
+  "\u315d",
+  "\u3149",
+  "\u11d3",
+  "\u1162",
+  "\u3146",
+  "\u118f",
+  "\ud7b1",
+  "\u11a4",
+  "\ud7f4",
+  "\ua962",
+  "\u1177",
+  "\u11b7",
+  "\ud7c3",
+  "\u1139",
+  "\ua96a",
+  "\ua97b",
+  "\u110b",
+  "\u119c",
+  "\u11f6",
+  "\u11aa",
+  "\u3170",
+  "\u11f8",
+  "\u3142",
+  "\u11bb",
+  "\u117d",
+  "\u11c3",
+  "\ud7fe",
+  "\u11ac",
+  "\u112b",
+  "\u1109",
+  "\u117a",
+  "\u11eb",
+  "\u1102",
+  "\u1191",
+  "\u1115",
+  "\u3172",
+  "\ud7d2",
+  "\u3151",
+  "\u1199",
+  "\ud7bb",
+  "\u11fb",
+  "\u11b5",
+  "\u1104",
+  "\u3161",
+  "\ua960",
+  "\u111c",
+  "\u11b2",
+  "\u1169",
+  "\u3175",
+  "\u11a0",
+  "\ud7f6",
+  "\u1166",
+  "\u1167",
+  "\u1193",
+  "\ud7eb",
+  "\u119a",
+  "\u1156",
+  "\u3152",
+  "\u1127",
+  "\u315f",
+  "\u115e",
+  "\ua97f",
+  "\u11a8",
+  "\u1194",
+  "\u112c",
+  "\u1180",
+  "\u1107",
+  "\u1152",
+  "\ua97a",
+  "\u1125",
+  "\u111e",
+  "\u112e",
+  "\u3157",
+  "\u1136",
+  "\u11d5",
+  "\u11ae",
+  "\ud7f8",
+  "\u112d",
+  "\u1189",
+  "\ua96f",
+  "\u118e",
+  "\u1154",
+  "\ud7f5",
+  "\u315e",
+  "\u11b4",
+  "\u318a",
+  "\u1183",
+  "\u1175",
+  "\u11ed",
+  "\u11ef",
+  "\u1105",
+  "\u1182",
+  "\ud7ea",
+  "\u1192",
+  "\u11d0",
+  "\ua978",
+  "\u1128",
+  "\u115f",
+  "\u3140",
+  "\u1133",
+  "\u111f",
+  "\u11bf",
+  "\u3134",
+  "\ud7ed",
+  "\u117f",
+  "\ud7ba",
+  "\u110a"
+]);
+
+const INVISIBLE_CHARACTER_JOINERS = new Set([
```

Comment:
A few ideas here:

Could we please document:

Why we've created this Set. Ex: Why are we doing something special for Invisible Character Joiners?
How we determined the values of this set. Ex: through what data source / process did we produce these values?
Also, what do you think about moving these values into a ".json" file under the /data directory? I think it's nice if we consolidate ideas like this together.

Reply:
Created json. Added this note: This function also handles invisible characters within graphemes, ensuring they are split into separate graphemes for better clarity.